specs/agents/tracing-transaction-grouping.md

```diff
@@ -0,0 +1,37 @@
+## Transaction Grouping
+
+Even though agents should choose a transaction name that has a reasonable cardinality,
+they can't always guarantee that.
+For example,
+when the auto-instrumentation of a job scheduling framework sets the transaction name to the name of the instrumented job,
+the agent has no control over the job name itself.
+While usually the job name is expected to have low cardinality,
+users might set dynamic parts as part of the job name, such as a UUID.
+
+In order to give users an option to group transactions whose name contain dynamic parts that don't require code changes,
+agents MAY implement the following configuration option:
+
+### `transaction_name_groups` configuration
```

Comment:
I considered making this option work generically on both transactions and spans. But in the majority of cases, you just want to group transactions. Applying the same wildcard matchers to spans, too would cause unnecessary overhead.

If we need such a feature for spans, too, we can consider adding another config options. Maybe even with the option to filter by span type/subtype.